packages/api/src/command/medical/document/document-bulk-download.ts

```diff
@@ -0,0 +1,170 @@
+import { Config } from "../../../shared/config";
+import { searchDocuments } from "../../../external/fhir/document/search-documents";
+import { chunk } from "lodash";
+import { processPatientDocumentRequest, MAPIWebhookStatus } from "./document-webhook";
+import { DocumentBulkDownloadDTO } from "../../../routes/medical/dtos/document-bulk-downloadDTO";
+
+import {
+  DocumentBulkDownloadProgress,
+  isDocBulkDownloadProcessing,
+  DocumentDownloadStatus,
+} from "../../../domain/medical/document-bulk-download";
+//import { getSignedUrls } from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+import { Util } from "../../../shared/util";
+import { getPatientOrFail } from "../patient/get-patient";
+import { Patient } from "../../../domain/medical/patient";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { storeBulkDownloadQueryInit } from "../patient/query-init";
+import { appendDocBulkDownloadProgress } from "../patient/append-bulk-doc-download-progress";
+import { DocumentReference } from "@medplum/fhirtypes";
+import { makeLambdaClient } from "../../../external/aws/lambda";
+import { getLambdaResultPayload } from "@metriport/core/external/aws/lambda";
+import {
+  DocumentBulkSignerLambdaRequest,
+  DocumentBulkSignerLambdaResponse,
+} from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+
+const BATCH_SIZE = 100;
+const lambdaClient = makeLambdaClient();
+const bulkSigningLambdaName = Config.getBulkUrlSigningLambdaName();
+
+/**
+ * The function `triggerBulkUrlSigning` triggers the bulk signing process lambda for a patient's documents and
+ * returns the progress of the bulk download. It also triggers the webhook for the bulk download.
+ * @param {string} cxId - cxId
+ * @param {string} patientId - patientId
+ * @returns a Promise that resolves to a DocumentBulkDownloadProgress object.
+ */
+export const startBulkGetDocumentUrls = async (
+  cxId: string,
+  patientId: string
+): Promise<DocumentBulkDownloadProgress> => {
+  if (!bulkSigningLambdaName) throw new Error("Bulk Signing Lambda Name is undefined");
+  const { log } = Util.out(`triggerBulkSigning - M patient ${patientId}`);
+
+  const patient = await getPatientOrFail({ id: patientId, cxId });
+
+  const docBulkDownloadProgress = patient.data.documentBulkDownloadProgress;
+  const requestId = getOrGenerateRequestId(docBulkDownloadProgress);
+
+  if (isDocBulkDownloadProcessing(docBulkDownloadProgress?.download)) {
+    log(
+      `Patient ${patientId}, Request ${requestId}, docBulkDownloadProgress is already 'processing', skipping...`
+    );
+    return createBulkDownloadQueryResponse("processing", patient);
+  }
+
+  const documents = await searchDocuments({ cxId, patientId });
```

Comment:
one concern here is that we are returning all documents for a given patient for each request. Maybe v2 we update this so that like documentQuery it only returns net new urls?